cpp/include/DataStorm/InternalT.h

```diff
@@ -439,13 +439,18 @@ namespace DataStormI
 
         template<typename FF> void init(const std::string& name, FF&& lambda)
         {
-            _name = name;
-            _lambda = std::forward<FF>(lambda);
+            std::unique_lock lock(_mutex);
```

Comment:
I’m thinking the mutex is probably not required, the factory is owned by the topic instance and the instances cannot be shared between threads

Reply:
If it's not required, you should add a comment instead of a mutex. And clearly the other functions (getName etc.) don't use a mutex.